### PR TITLE
Add support for facility teleconsultation info sync resource

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/878/add-support-for-facility-teleconsultation-info-sync-resource

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationSync.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.simple.clinic.main.TypedPreference.Type.LastTeleconsultationFacilityPullToken
+import org.simple.clinic.sync.ModelSync
+import org.simple.clinic.sync.SyncConfig
+import org.simple.clinic.sync.SyncCoordinator
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.Optional
+import javax.inject.Inject
+import javax.inject.Named
+
+class TeleconsultationSync @Inject constructor(
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationSync.kt:16]: We need to add `TeleconsultationSync` class to the `SyncModule#syncs` provider method for the sync to get triggered.

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good. Left few comments. PTAL

File: app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    )
+
+    val expectedTeleconsultationFacilityWithMedicalOfficers = listOf(
+        TestData.teleconsultationFacilityWithMedicalOfficers(
+            teleconsultationFacilityInfo = teleconsultationFacility1,
+            medicalOfficers = listOf(medicalOfficer1InFacility1, medicalOfficer2)
+        ),
+        TestData.teleconsultationFacilityWithMedicalOfficers(
+            teleconsultationFacilityInfo = teleconsultationFacility2,
+            medicalOfficers = listOf(medicalOfficer1InFacility2)
```
[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt:84]: Both `medicalOfficer1InFacility1` & `medicalOfficer1InFacility2` are same. But I have made them explicit to make sure it's easier to test and verify. Also to verify the single with MO assigned to multiple facilities case.

[the remark this answers is not in the document]
Comment (Honey14) [reply app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt:84]: We can instead use `medicalOfficerInFacility1` here as well?

[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt:84]: Yes, replaced it with that

[the remark this answers is not in the document]
Comment (Honey14) [reply app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt]: Can we also do the same in `TeleconsultationFacilityRepositoryAndroidTest`?

[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/androidTest/java/org/simple/clinic/summary/teleconsultation/sync/TeleconsultationFacilityWithMedicalOfficersTest.kt]: What I wanted to bring up was that the explicit nature of this is being taken care of when we construct the cross reference model.

```kotlin
val medicalOfficer1 = /* */
val medicalOfficer2 = /* */

val facility1 = /* */
val facility2 = /* */

val teleconsultInfoForFacility1 = TestData.crossRef(facility1, list(medicalOfficer1, medicalOfficer2))
val teleconsultInfoForFacility2 = TestData.crossRef(facility2, list(medicalOfficer2))
```

**Review by Honey14 — APPROVED** [review]
Looks great

**Pull request merged**